### PR TITLE
fix(nasdaq): fetch index based on last weekday (#22)

### PR DESCRIPTION
**Pull request by AugustSoderlund**
Changed in `nasdaq_requests.py` to allow the date where the weighting of the index is retrieved to the last weekday.

Did this by calculating using `timedelta` with the difference between todays day and the latest weekday.

### CODE REVIEW

**Comment by wilhelmagren** [comment]
Also added some branch protection rules to main:
- Require a pull request before merging 
  - Require approvals 
- Require status checks to pass before merging 
- Require conversation resolution before merging 
- Require signed commits 

thoughts? @AugustSoderlund

File: finq/datautil/nasdaq_requests.py
Comment (wilhelmagren) [inline finq/datautil/nasdaq_requests.py:78]: Very nice.

**Review by wilhelmagren — APPROVED** [review]

File: finq/datautil/nasdaq_requests.py
Comment (AugustSoderlund) [reply finq/datautil/nasdaq_requests.py:78]: thanks 😄

**Comment by AugustSoderlund** [comment]
> Also added some branch protection rules to main:
> 
> * Require a pull request before merging
>   
>   * Require approvals
> * Require status checks to pass before merging
> * Require conversation resolution before merging
> * Require signed commits
> 
> thoughts? @AugustSoderlund

This sounds good! 👍

**Comment by wilhelmagren** [comment]
I removed "Require signed commits"; i thought it only meant the latest commit had to be signed, but it meant ALL commits had to be signed. we can't do this, so lets suffice with the merge being signed is good enough :P

**Comment by AugustSoderlund** [comment]
> I removed "Require signed commits"; i thought it only meant the latest commit had to be signed, but it meant ALL commits had to be signed. we can't do this, so lets suffice with the merge being signed is good enough :P

Alright, it will probably suffice to have at least 1 reviewer.

**Pull request merged**
